crates/pallet-domains/src/lib.rs

```diff
@@ -1493,31 +1490,14 @@ impl<T: Config> Pallet<T> {
             FraudProofError::ChallengingGenesisReceipt
         );
 
-        if let FraudProof::InvalidBundle(InvalidBundleProof::TotalRewards(
-            InvalidTotalRewardsProof {
-                domain_id,
-                block_number,
-                domain_block_hash,
-                storage_proof,
-                ..
-            },
-        )) = fraud_proof
+        if let FraudProof::InvalidTotalRewards(InvalidTotalRewardsProof { storage_proof, .. }) =
+            fraud_proof
         {
-            let domain_number = T::DomainNumber::decode(&mut block_number.encode().as_slice())
-                .map_err(|_| FraudProofError::FailedToDecodeDomainBlockNumber)?;
-            let domain_block_hash =
-                T::DomainHash::decode(&mut domain_block_hash.encode().as_slice())
-                    .map_err(|_| FraudProofError::FailedToDecodeDomainBlockHash)?;
-            let state_root = T::Hash::decode(
-                &mut StateRoots::<T>::get((domain_id, domain_number, domain_block_hash))
-                    .ok_or(FraudProofError::MissingDomainStateRoot)?
-                    .encode()
-                    .as_slice(),
-            )
-            .map_err(|_| FraudProofError::FailedToDecodeDomainBlockHash)?;
-            let storage_key = StorageKey(
-                sp_domains::fraud_proof::OperatorBlockRewards::storage_value_final_key().to_vec(),
-            );
+            let state_root = bad_receipt.final_state_root.encode();
+            let state_root = T::Hash::decode(&mut state_root.as_slice())
+                .map_err(|_| FraudProofError::FailedToDecodeDomainBlockHash)?;
+            let storage_key =
+                StorageKey(sp_domains::fraud_proof::operator_block_rewards_final_key());
             let storage_proof = storage_proof.clone();
 
             let total_rewards = StorageProofVerifier::<T::Hashing>::verify_and_get_value::<
```

Comment:
`T::Hashing` seems also should be the domain hashing, #1824 has introduced `T::DomainHashing` if you don't want to add it in this PR plz leave a TODO

Reply:
That PR did not land yet. Leave a TODO there since there is not a blocker here.